include/linux/slub_def.h

```diff
@@ -185,6 +186,8 @@ static inline unsigned int __obj_to_index(const struct kmem_cache *cache,
 static inline unsigned int obj_to_index(const struct kmem_cache *cache,
 					const struct page *page, void *obj)
 {
+	if (is_kfence_address(obj))
```

Comment:
Hm, the same could be done here, but there are multiple callers of obj_to_index(), and we probably don't want to open-code them all.
I wonder if this doesn't violate any contracts about the index being translatable back to the object address.

Reply:
We should push for clean code first. If somebody wants ugly code, we can make it ugly. ;-)

Re violating contracts: For slab.c index_to_obj() works because we set s_mem correctly. Other than those in slab.c, I don't see any reverse translations.

And I also see we can probably get rid of the is_kfence_address() guard in slab.c's __check_heap_object() for the obj_to_index() translation (it just returns 0). I'll add that commit. So evidently, things before these fixes were definitely more broken.